src/serde_xml_plist.rs

```diff
@@ -1,467 +1,657 @@
-//! Helpers for deserializing plist values from untyped XML
+//! Helpers for (de)serializing plist values from untyped XML
 //!
 //! This is essentially glue between the `plist` crate and the `quick_xml` crate.
 //! It allows plist values, dictionaries and arrays to be used inside types that
 //! derive Deserialize/Serialize.
 
-use std::{fmt::Display, marker::PhantomData, str::FromStr};
-
 use base64::{engine::general_purpose::STANDARD as base64_standard, Engine};
 use plist::{Dictionary, Value};
-use serde::{
-    de::{Error as DeError, Visitor},
-    Deserialize, Deserializer,
-};
+use serde::ser::SerializeStruct;
+use serde::{Deserializer, Serializer};
 
 /// Deserialize a plist Dictionary
 ///
 /// This relies on the specific structure presented by the quick_xml crate and
 /// is likely not suited to other formats.
-pub fn deserialize_dict<'de, D>(deserializer: D) -> Result<Dictionary, D::Error>
+pub(crate) fn deserialize<'de, D>(deserializer: D) -> Result<Dictionary, D::Error>
 where
     D: Deserializer<'de>,
 {
-    deserializer.deserialize_map(ValueVisitor::DictOnly).map(|x| x.into_dictionary().unwrap())
+    deserializer.deserialize_map(de::ValueVisitor::DictOnly).map(|x| x.into_dictionary().unwrap())
 }
 
-struct DictWrapper(Dictionary);
-struct ValueWrapper(Value);
-struct ArrayWrapper(Vec<Value>);
-struct IntWrapper(plist::Integer);
-
-/// The literal keyword 'key'.
-struct KeyKeywordLiteral;
+mod de {
```

Comment:
just an observation: I agree it makes sense to go with the ser/de mods, but moving all the existing code into a new module in this PR makes it very difficult to tell where the actual changes are, versus just changes in indent.

No need to change this now, but it's something to consider when making this kind of change, in the future. I would have probably left the existing code where it was, and then added a shim `de` module that just used the types that were declared in the module root; after the PR was merged it's an easy refactor to move things around. :)

Reply:
oh cool, TIL. Will look into that setting...